torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -96,6 +103,15 @@ template <typename T> void atomic_add(volatile T *addr, T offset) {
                                                std::memory_order_relaxed));
 }
 
+// Currently C++17 is used so float is still backed by atomic_floating_add
+// WARN: If switching to C++20, please pay attention to
+// microbench inductor_cpu_atomic.py performance
+#if __cplusplus < 201711L // __cpp_lib_atomic_float
+template <> void atomic_add<float>(volatile float *addr, float offset) { atomic_floating_add(addr, offset); }
+template <> void atomic_add<double>(volatile double *addr, double offset) { atomic_floating_add(addr, offset); }
+#endif
```

Comment:
Let's hold the floating point support with c++20 considering we don't see perf gain.